libwallet/assets_manager.go

```diff
@@ -992,11 +1008,82 @@ func (mgr *AssetsManager) PrepareDexSupportForDcrWallet() error {
 			return nil, fmt.Errorf("DEX wallet not supported for %s", walletParams.Name)
 		}
 
-		return dcr.NewDEXWallet(dcrAsset.Wallet, dcrAsset.SyncData()), nil
+		return dcr.NewDEXWallet(dcrAsset, accountNumber, dcrAsset.SyncData()), nil
 	}
 
 	err := dexDcr.RegisterCustomWallet(walletMaker, def)
-	if err != nil && !strings.Contains(err.Error(), "already support" /* this is part of the error string returned if we've already registered a customer wallet */) {
+	if err != nil && !errors.Is(err, dexasset.ErrWalletTypeAlreadyRegistered) { // this is the only error from RegisterCustomWallet, ignore?
+		return err
+	}
```

Comment:
This should now panic upstream. Use an atomic variable to prevent multiple registrations here.